pkg/framework/framework.go

```diff
@@ -296,6 +296,26 @@ func SkipIfNotTechPreviewNoUpgrade(oc *gatherer.CLI, cl runtimeclient.Client) {
 	}
 }
 
+// SkipIfNotCustomNoUpgrade skip test if a cluster is not a CustomNoUpgrade cluster.
+func SkipIfNotCustomNoUpgrade(oc *gatherer.CLI, cl runtimeclient.Client) {
+	featureSet, err := oc.WithoutNamespace().Run("get").Args("featuregate", "cluster", "-o=jsonpath={.spec.featureSet}").Output()
+	Expect(err).NotTo(HaveOccurred(), "Failed to get featureSet")
+
+	if featureSet != string(configv1.CustomNoUpgrade) {
+		Skip("FeatureSet is not CustomNoUpgrade, skip it!")
+	}
+}
+
+// SkipIfNotTechPreviewOrCustomNoUpgrade skip test if a cluster does not have TechPreviewNoUpgrade or CustomNoUpgrade featuregate enabled.
+func SkipIfNotTechPreviewOrCustomNoUpgrade(oc *gatherer.CLI, cl runtimeclient.Client) {
+	featureSet, err := oc.WithoutNamespace().Run("get").Args("featuregate", "cluster", "-o=jsonpath={.spec.featureSet}").Output()
+	Expect(err).NotTo(HaveOccurred(), "Failed to get featureSet")
+
+	if featureSet != string(configv1.TechPreviewNoUpgrade) && featureSet != string(configv1.CustomNoUpgrade) {
+		Skip("Cluster does not have TechPreviewNoUpgrade or CustomNoUpgrade featuregate enabled. Not a valid cluster for this test.")
+	}
+}
```

Comment:
we already have SkipIfNotTechPreviewNoUpgrade, maybe keep SkipIfNotCustomNoUpgrade only?